From e9ed329e339f12d4c73bcd5cffed5df9c17fc9f9 Mon Sep 17 00:00:00 2001
From: Martin Rehfeld <martin.rehfeld@glnetworks.de>
Date: Thu, 11 Sep 2008 11:56:51 +0200
Subject: [PATCH] added failing tests
 * AssetTagHelperTest#test_caching_javascript_include_tag_with_relative_url_root
 * AssetTagHelperTest#test_caching_stylesheet_include_tag_with_relative_url_root

---
 actionpack/test/template/asset_tag_helper_test.rb |   50 +++++++++++++++++++++
 1 files changed, 50 insertions(+), 0 deletions(-)

diff --git a/actionpack/test/template/asset_tag_helper_test.rb b/actionpack/test/template/asset_tag_helper_test.rb
index 7e40a55..fe127da 100644
--- a/actionpack/test/template/asset_tag_helper_test.rb
+++ b/actionpack/test/template/asset_tag_helper_test.rb
@@ -385,6 +385,31 @@ class AssetTagHelperTest < ActionView::TestCase
   ensure
     FileUtils.rm_f(File.join(ActionView::Helpers::AssetTagHelper::JAVASCRIPTS_DIR, 'combined.js'))
   end
+  
+  def test_caching_javascript_include_tag_with_relative_url_root
+    ENV["RAILS_ASSET_ID"] = ""
+    ActionController::Base.relative_url_root = "/collaboration/hieraki"
+    ActionController::Base.perform_caching = true
+
+    assert_dom_equal(
+      %(<script src="/collaboration/hieraki/javascripts/all.js" type="text/javascript"></script>),
+      javascript_include_tag(:all, :cache => true)
+    )
+
+    assert File.exist?(File.join(ActionView::Helpers::AssetTagHelper::JAVASCRIPTS_DIR, 'all.js'))
+
+    assert_dom_equal(
+      %(<script src="/collaboration/hieraki/javascripts/money.js" type="text/javascript"></script>),
+      javascript_include_tag(:all, :cache => "money")
+    )
+
+    assert File.exist?(File.join(ActionView::Helpers::AssetTagHelper::JAVASCRIPTS_DIR, 'money.js'))
+
+  ensure
+    ActionController::Base.relative_url_root = nil
+    FileUtils.rm_f(File.join(ActionView::Helpers::AssetTagHelper::JAVASCRIPTS_DIR, 'all.js'))
+    FileUtils.rm_f(File.join(ActionView::Helpers::AssetTagHelper::JAVASCRIPTS_DIR, 'money.js'))
+  end
 
   def test_caching_javascript_include_tag_when_caching_off
     ENV["RAILS_ASSET_ID"] = ""
@@ -456,6 +481,31 @@ class AssetTagHelperTest < ActionView::TestCase
     FileUtils.rm_f(File.join(ActionView::Helpers::AssetTagHelper::STYLESHEETS_DIR, 'styles.css'))
   end
 
+  def test_caching_stylesheet_link_tag_with_relative_url_root
+    ENV["RAILS_ASSET_ID"] = ""
+    ActionController::Base.relative_url_root = "/collaboration/hieraki"
+    ActionController::Base.perform_caching = true
+
+    assert_dom_equal(
+      %(<link href="/collaboration/hieraki/stylesheets/all.css" media="screen" rel="stylesheet" type="text/css" />),
+      stylesheet_link_tag(:all, :cache => true)
+    )
+
+    expected = Dir["#{ActionView::Helpers::AssetTagHelper::STYLESHEETS_DIR}/*.css"].map { |p| File.mtime(p) }.max
+    assert_equal expected, File.mtime(File.join(ActionView::Helpers::AssetTagHelper::STYLESHEETS_DIR, 'all.css'))
+
+    assert_dom_equal(
+      %(<link href="/collaboration/hieraki/stylesheets/money.css" media="screen" rel="stylesheet" type="text/css" />),
+      stylesheet_link_tag(:all, :cache => "money")
+    )
+
+    assert File.exist?(File.join(ActionView::Helpers::AssetTagHelper::STYLESHEETS_DIR, 'money.css'))
+  ensure
+    ActionController::Base.relative_url_root = nil
+    FileUtils.rm_f(File.join(ActionView::Helpers::AssetTagHelper::STYLESHEETS_DIR, 'all.css'))
+    FileUtils.rm_f(File.join(ActionView::Helpers::AssetTagHelper::STYLESHEETS_DIR, 'money.css'))
+  end
+
   def test_caching_stylesheet_include_tag_when_caching_off
     ENV["RAILS_ASSET_ID"] = ""
     ActionController::Base.perform_caching = false
-- 
1.5.5.3

